pkg/controllers/jobset_controller.go

```diff
@@ -126,6 +127,55 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, nil
 	}
 
+	// If JobSpec has a suspend field of true,
+	// set condition to say jobspec is suspended.
+	if js.Spec.Suspend != nil && *js.Spec.Suspend && !JobSetSuspended(&js) {
+		if err := r.updateStatusWithCondition(ctx, &js, corev1.EventTypeNormal, metav1.Condition{
```

Comment:
We can depend on the job status because that will cause race conditions: imagine that case where this update succeeded, but then the controller restarts, we will not get to suspending the jobs ever.

Imagine also the case where a child job was unsuspended out of band, we need to suspend it again.

The condition above should only consider the js.spec.suspend field, `!JobSetSuspended(&js)` must not be part of the condition to determine where or not to suspend/unsuspend the child jobs.

Reply:
I don't really understand this.  

We check for JobSetFinished or JobSetFailed by checking the condition.

I tried to check suspend via the spec but I realize I can't really distinguish between a JobSet starting or a JobSet being resumed.  I know that most of the tests fail because we are pushing out a condition that says a jobset is resumed even if we are starting the job without a suspend.  

I check the suspend field and then check the status of the jobset to see if the jobset was previously suspended.  If that is the case then we push a condition that says we were resumeded. 

Otherwise I am unsure how to detect between suspend of false and suspend of false when it was original suspend of true.